sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -137,13 +180,160 @@ class GpuWindowExecMeta(windowExec: WindowExec,
   override def getResultColumnsOnly: Boolean = resultColumnsOnly
 }
 
-case class GpuWindowExec(
-    windowExpressionAliases: Seq[Expression],
-    partitionSpec: Seq[Expression],
-    orderSpec: Seq[SortOrder],
-    child: SparkPlan,
-    resultColumnsOnly: Boolean
-  ) extends UnaryExecNode with GpuExec {
+object GpuWindowExec extends Arm {
+  def isRunningWindow(spec: GpuWindowSpecDefinition): Boolean = spec match {
+    case GpuWindowSpecDefinition(_, _, GpuSpecifiedWindowFrame(RowFrame,
+    GpuSpecialFrameBoundary(UnboundedPreceding), GpuSpecialFrameBoundary(CurrentRow))) => true
+    case _ => false
+  }
+
+  def fixerIndexMap(windowExpressionAliases: Seq[Expression]): Map[Int, BatchedRunningWindowFixer] =
+    windowExpressionAliases.zipWithIndex.flatMap {
+      case (GpuAlias(GpuWindowExpression(func, _), _), index) =>
+        func match {
+          case f: GpuBatchedRunningWindowFunction[_] =>
+            Some((index, f.newFixer()))
+          case GpuAggregateExpression(f: GpuBatchedRunningWindowFunction[_], _, _, _, _) =>
+            Some((index, f.newFixer()))
+          case _ => None
+        }
+      case _ => None
+    }.toMap
+
+  def computeRunningNoPartitioning(
+      iter: Iterator[ColumnarBatch],
+      boundProjectList: Seq[GpuExpression],
+      numOutputBatches: GpuMetric,
+      numOutputRows: GpuMetric,
+      opTime: GpuMetric): Iterator[ColumnarBatch] = {
+    val fixers = fixerIndexMap(boundProjectList)
+    TaskContext.get().addTaskCompletionListener[Unit](_ => fixers.values.foreach(_.close()))
+
+    iter.map { cb =>
+      val numRows = cb.numRows
+      numOutputBatches += 1
+      numOutputRows += numRows
+      withResource(new MetricRange(opTime)) { _ =>
+        withResource(GpuProjectExec.projectAndClose(cb, boundProjectList, NoopMetric)) { full =>
+          closeOnExcept(ArrayBuffer[ColumnVector]()) { newColumns =>
+            boundProjectList.indices.foreach { idx =>
+              val column = full.column(idx).asInstanceOf[GpuColumnVector]
+              val fixer = fixers.get(idx)
+              if (fixer.isDefined) {
```

Comment:
nit, would be nice to keep `fixer.get` in a val to reuse in the couple of places utilized here.